fxa-client/src/config.rs

```diff
@@ -130,51 +144,51 @@ impl Config {
     }
 
     pub fn auth_url(&self) -> Result<Url> {
-        Url::parse(&self.remote_config.auth_url).map_err(|e| e.into())
+        Url::parse(&self.remote_config()?.auth_url).map_err(|e| e.into())
```

Comment:
Is there a reason to do `blah.map_err(|e| e.into())` over `Ok(blah?)`?

Reply:
I remember a previous review where I was told `Ok(blah?)` is bad, I'd be happy to switch back if we can have a consensus on that subject.